modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js

```diff
@@ -140,17 +147,36 @@ export default function Chart({
 
 	const [, addHistoricalWarning] = useHistoricalWarning();
 
-	const {actions, state: chartState} = useChartState({
-		publishDate,
-		timeSpanKey,
-	});
+	const chartState = useChartState();
+
+	const addDataSetItems = useAddDataSetItems();
+
+	const setLoading = useSetLoading();
+
+	const changeTimeSpanKey = useChangeTimeSpanKey();
+
+	const previousTimeSpan = usePreviousTimeSpan();
+
+	const nextTimeSpan = useNextTimeSpan();
+
+	const isPreviousPeriodButtonDisabled = useIsPreviousPeriodButtonDisabled();
+
+	const dateFormatters = useMemo(() => dateFormat(languageTag), [
+		languageTag,
+	]);
+
+	const {firstDate, lastDate} = useDateTitle();
+
+	const title = useMemo(() => {
+		return dateFormatters.formatChartTitle([firstDate, lastDate]);
+	}, [dateFormatters, firstDate, lastDate]);
```

Comment:
Just a due diligience question: are you sure the computational cost that you're saving with `useMemo` here is greater than what you're spending? (I don't know how expensive `dateFormat()` and `formatChartTitle()` are.)

Reply:
I think maybe I can create a variable just calling `dateFormatters.formatCharTitle` without the memoization. `dateFormat` is just the initialization of a set of methods to transform data with a specific format and `formatChartTitle` is inside that methods and inside, it uses `Intl.DateTimeFormat` with specific properties to represent it numeric, string, etc. I think it's not super expensive... 🤔 Also, I create a new hook to get the `firstDate` and `lastDate` to calculate the chart title (see `useDateTitle). So maybe we can remove this `useMemo`. What do you think?